### PR TITLE
chore: remove aspectj-rt from the library

### PR DESCRIPTION
**Pull request by jeromevdl**
**Issue #, if available:** #1401

## Description of changes:

- remove aspectj-rt from the library itself => requires to add it in the project that uses powertools, but at least developers can choose their version and are not stick to 1.9.7 which does not work anymore with java 17.
- update example to simplify the configuration (just set the aspectj version in the profile and keep the same plugin configuration). If we remove java8, we can remove the profile (default version of aspectj is 1.9.20)

**Checklist**

 

* [x] [Meet tenets criteria](https://docs.powertools.aws.dev/lambda-java/#tenets)
* [x] Update tests
* [ ] Update docs ==> not yet
* [x] PR title follows [conventional commit semantics](https://www.conventionalcommits.org/en/v1.0.0/)

## Breaking change checklist

requires the developers to add to their pom
```xml 
        <dependency>
            <groupId>org.aspectj</groupId>
            <artifactId>aspectjrt</artifactId>
            <version>...</version> <!-- 1.9.7 for java 8 or higher otherwise -->
        </dependency>
```

**RFC issue #**:

* [ ] Migration process documented ==> not yet
* [ ] Implement warnings (if it can live side by side)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by jeromevdl** [comment]
@roamingthings, what do you think about it?

**Review by roamingthings — COMMENTED** [review]
The parent pom still uses 1.9.7 and I'm not sure if this would be treated as a transient dependency. I will try to do some testing.

**Comment by scottgerring** [comment]
I haven't had a chance to think about this in depth yet, but my initial reaction is:

>  requires to add it in the project that uses powertools

Where does this fall on the spectrum of _breaking changes_? We'd have to at the very least message this clearly in the release notes, as it is likely to break most client's builds when they roll their PT dep(s) forward.

**Comment by roamingthings** [comment]
> as it is likely to break most client's builds when they roll their PT dep(s) forward
This relates to all project that don't use the AspectJ plugin.

**Comment by jeromevdl** [comment]
> The parent pom still uses 1.9.7 and I'm not sure if this would be treated as a transient dependency. I will try to do some testing.

@roamingthings, the parent pom references it but no module actually uses it, and we still need it for our own code. And since we support java8 (until @scottgerring decides to leave it 😄), we need to keep the 1.9.7 version. But it should not be retrieved in your application. Actually you need to explicitly specify one version (1.9.20 most probably as of now). I didn't try with Gradle yet...

**Comment by roamingthings** [comment]
@jeromevdl 
> I didn't try with Gradle yet...

That's what I meant by
> I will try to do some testing.

**Comment by jeromevdl** [comment]
moving as draft, I want to merge main into v2 to get the gradle sample working with this

**Review by scottgerring — APPROVED** [review]
LGTM. And I am super glad we have the E2E tests for this sort of thing.

File: examples/powertools-examples-batch/pom.xml [hunk 1]
```diff
@@ -132,6 +145,7 @@
               - If you are running on Java 1.8, you should apply the aspectJ version here to the project, and remove
                 the profile.
```
[the remark this answers is not in the document]
Comment (jeromevdl) [reply examples/powertools-examples-batch/pom.xml:153]: it is, as we have to specify a lower version for aspectj-rt

**Pull request merged**